apps/web/modules/onboarding/organization/details/validated-organization-slug.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+
+import { useCallback, useEffect, useRef, useState, useTransition } from "react";
+
+import classNames from "@calcom/ui/classNames";
+import { Label, TextField } from "@calcom/ui/components/form";
+import { Icon } from "@calcom/ui/components/icon";
+
+import { checkSlugAvailability } from "./action/check-slug-availability";
+
+type ValidationState = "idle" | "checking" | "available" | "taken";
+
+type ValidatedOrganizationSlugProps = {
+  value: string;
+  onChange: (value: string) => void;
+  onValidationChange?: (isValid: boolean) => void;
+};
+
+export function ValidatedOrganizationSlug({
+  value,
+  onChange,
+  onValidationChange,
+}: ValidatedOrganizationSlugProps) {
+  const [validationState, setValidationState] = useState<ValidationState>("idle");
+  const [errorMessage, setErrorMessage] = useState<string>("");
+  const [_isPending, startTransition] = useTransition();
+  const timeoutRef = useRef<NodeJS.Timeout>();
+
+  const validateSlug = useCallback(
+    (slug: string) => {
+      if (!slug || slug.trim() === "") {
+        setValidationState("idle");
+        setErrorMessage("");
+        onValidationChange?.(false);
+        return;
+      }
+
+      setValidationState("checking");
+
+      // Call server action
+      checkSlugAvailability(slug).then((result) => {
+        startTransition(() => {
+          if (result.available) {
+            setValidationState("available");
+            setErrorMessage("");
+            onValidationChange?.(true);
+          } else {
+            setValidationState("taken");
+            setErrorMessage(result.message || "This slug is not available");
+            onValidationChange?.(false);
+          }
+        });
+      });
+    },
+    [onValidationChange]
+  );
+
+  useEffect(() => {
```

Comment:
nit: it'd be cleaner to use the existing `import { debounce } from "lodash";`

Reply:
Will tackle top of the stack! ty